Code/GraphMol/Resonance.cpp

```diff
@@ -1051,12 +1057,28 @@ void CEVect2::idxToDepthWidth(unsigned int idx, unsigned int &d,
 
 // get the pointer to the BondElectrons object for bond having index bi
 BondElectrons *ConjElectrons::getBondElectronsWithIdx(unsigned int bi) {
-  return d_conjBondMap[bi];
+  BondElectrons *be;
+  try {
+    be = d_conjBondMap.at(bi);
+  }
+  catch (const std::out_of_range& e) {
+    be = nullptr;
+  }
+
+  return be;
 }
 
 // get the pointer to the AtomElectrons object for atom having index ai
 AtomElectrons *ConjElectrons::getAtomElectronsWithIdx(unsigned int ai) {
-  return d_conjAtomMap[ai];
+  AtomElectrons *ae;
+  try {
+    ae = d_conjAtomMap.at(ai);
+  }
+  catch (const std::out_of_range& e) {
+    ae = nullptr;
+  }
+
+  return ae;
```

Comment:
```suggestion
  if (d_conjAtomMap.count(ai))
    return d_conjAtomMap[ai];
  else
    return nullptr;
```

Reply:
I agree that the `try..catch` block is not very elegant, and probably not the most efficient, but using `std::map::count()` and then `std::map::operator[]` will do two lookups. So I modified the code to use `std::map::find` instead, which is more elegant then `try..catch` and probably also more efficient :-) Thanks!